components/remote_settings/src/storage.rs

```diff
@@ -19,37 +22,75 @@ use crate::{Attachment, RemoteSettingsRecord, Result};
-///   - Maybe we should refactor this to use the DAO pattern like suggest does.
-pub struct Storage {}
+pub struct Storage {
+    conn: Arc<Mutex<Connection>>,
+}
 
 impl Storage {
-    pub fn new(_path: Utf8PathBuf) -> Result<Self> {
-        Ok(Self {})
+    pub fn new(path: Utf8PathBuf) -> Result<Self> {
+        let conn = Connection::open(path)?;
+        let storage = Self {
+            conn: Arc::new(Mutex::new(conn)),
+        };
+        storage.initialize_database()?;
+
+        Ok(storage)
+    }
+
+    // Create the different tables for records and attachements for every new sqlite path
+    fn initialize_database(&self) -> Result<()> {
+        let conn = self.conn.lock().unwrap();
+        conn.execute_batch(
+            "
+            CREATE TABLE IF NOT EXISTS records (
+                id TEXT PRIMARY KEY,
+                collection_url TEXT NOT NULL,
+                data BLOB NOT NULL
+            );
+            CREATE TABLE IF NOT EXISTS attachments (
+                id TEXT PRIMARY KEY,
+                collection_url TEXT NOT NULL,
+                data BLOB NOT NULL
+            );
+            ",
+        )?;
+        Ok(())
     }
 
     /// Get the last modified timestamp for the stored records
     ///
     /// Returns None if no records are stored or if `collection_url` does not match the
     /// `collection_url` passed to `set_records`.
-    pub fn get_last_modified_timestamp(&self, _collection_url: &str) -> Result<Option<u64>> {
-        Ok(None)
+    pub fn get_last_modified_timestamp(&self, collection_url: &str) -> Result<Option<u64>> {
+        let conn = self.conn.lock().unwrap();
+        let mut stmt = conn.prepare(
+            "SELECT MAX(json_extract(data, '$.last_modified')) FROM records WHERE collection_url = ?",
+        )?;
+
+        let result: Option<u64> = stmt.query_row(params![collection_url], |row| row.get(0))?;
+
+        Ok(result)
     }
 
     /// Get cached records for this collection
     ///
-    /// Returns None if no records are stored or if `collection_url` does not match the
-    /// `collection_url` passed to `set_records`.
-    pub fn get_records(&self, _collection_url: &str) -> Result<Option<Vec<RemoteSettingsRecord>>> {
-        Ok(None)
+    /// Returns None if no records are stored or if `collection_url` does not match the `collection_url` passed
+    /// to `set_records`.
+    pub fn get_records(&self, collection_url: &str) -> Result<Option<Vec<RemoteSettingsRecord>>> {
+        let conn = self.conn.lock().unwrap();
+        let mut stmt = conn.prepare("SELECT data FROM records WHERE collection_url = ?")?;
+        let mut rows = stmt.query(params![collection_url])?;
+        let mut records = Vec::new();
+        while let Some(row) = rows.next()? {
+            let data: Vec<u8> = row.get(0)?;
+            let record: RemoteSettingsRecord = serde_json::from_slice(&data)?;
+            records.push(record);
+        }
+        if records.is_empty() {
+            Ok(None)
```

Comment:
Like I mentioned above, this one feels a bit risky to me since there could be a collection with 0 records.